src/Evolution/DiscontinuousGalerkin/SlopeLimiters/Hierarchical.hpp

```diff
@@ -0,0 +1,982 @@
+ *          \alpha_2\left(c^{l,m,n}_{0,1,0}-c^{l,m-1,n}_{0,1,0}\right)
+ *     \right),\\
+ * \tilde{c}^{l,m,n}_{0,0,2}=\mathrm{minmod}
+ *   &\left(c_{0,0,2}^{l,m,n},
+ *          \alpha_2\left(c^{l,m,n+1}_{0,0,1}-c^{l,m,n}_{0,0,1}\right),
+ *          \alpha_2\left(c^{l,m,n}_{0,0,1}-c^{l,m,n-1}_{0,0,1}\right)
+ *     \right),\\
+ * \tilde{c}^{l,m,n}_{1,1,1}=\mathrm{minmod}
+ *   &\left(c_{1,1,1}^{l,m,n},
+ *          \alpha_1\left(c^{l+1,m,n}_{0,1,1}-c^{l,m,n}_{0,1,1}\right),
+ *          \alpha_1\left(c^{l,m,n}_{0,1,1}-c^{l-1,m,n}_{0,1,1}\right),
+ *     \right.\\
+ *   &\;\;\;\;
+ *          \alpha_1\left(c^{l,m+1,n}_{1,0,1}-c^{l,m,n}_{1,0,1}\right),
+ *          \alpha_1\left(c^{l,m,n}_{1,0,1}-c^{l,m-1,n}_{1,0,1}\right),\\
+ *   &\;\;\;\;\left.
+ *          \alpha_1\left(c^{l,m,n+1}_{1,1,0}-c^{l,m,n}_{1,1,0}\right),
+ *          \alpha_1\left(c^{l,m,n}_{1,1,0}-c^{l,m,n-1}_{1,1,0}\right)
+ *     \right),\\
+ * \tilde{c}^{l,m,n}_{1,1,0}=\mathrm{minmod}
+ *   &\left(c_{1,1,0}^{l,m,n},
+ *          \alpha_1\left(c^{l+1,m,n}_{0,1,0}-c^{l,m,n}_{0,1,0}\right),
+ *          \alpha_1\left(c^{l,m,n}_{0,1,0}-c^{l-1,m,n}_{0,1,0}\right),
+ *     \right.\\
+ *   &\;\;\;\;\left.
+ *          \alpha_1\left(c^{l,m+1,n}_{1,0,0}-c^{l,m,n}_{1,0,0}\right),
+ *          \alpha_1\left(c^{l,m,n}_{1,0,0}-c^{l,m-1,n}_{1,0,0}\right),
+ *     \right),\\
+ * \tilde{c}^{l,m,n}_{1,0,1}=\mathrm{minmod}
+ *   &\left(c_{1,0,1}^{l,m,n},
+ *          \alpha_1\left(c^{l+1,m,n}_{0,0,1}-c^{l,m,n}_{0,0,1}\right),
+ *          \alpha_1\left(c^{l,m,n}_{0,0,1}-c^{l-1,m,n}_{0,0,1}\right),
+ *     \right.\\
+ *   &\;\;\;\;\left.
+ *          \alpha_1\left(c^{l,m,n+1}_{1,0,0}-c^{l,m,n}_{1,0,0}\right),
+ *          \alpha_1\left(c^{l,m,n}_{1,0,0}-c^{l,m,n-1}_{1,0,0}\right)
+ *     \right),\\
+ * \tilde{c}^{l,m,n}_{0,1,1}=\mathrm{minmod}
+ *   &\left(c_{0,1,1}^{l,m,n},
+ *          \alpha_1\left(c^{l,m+1,n}_{0,0,1}-c^{l,m,n}_{0,0,1}\right),
+ *          \alpha_1\left(c^{l,m,n}_{0,0,1}-c^{l,m-1,n}_{0,0,1}\right),
+ *   \right.\\
+ *   &\;\;\;\;\left.
+ *          \alpha_1\left(c^{l,m,n+1}_{0,1,0}-c^{l,m,n}_{0,1,0}\right),
+ *          \alpha_1\left(c^{l,m,n}_{0,1,0}-c^{l,m,n-1}_{0,1,0}\right)
+ *     \right),\\
+ * \tilde{c}^{l,m,n}_{1,0,0}=\mathrm{minmod}
+ *   &\left(c_{1,0,0}^{l,m,n},
+ *          \alpha_1\left(c^{l+1,m,n}_{0,0,0}-c^{l,m,n}_{0,0,0}\right),
+ *          \alpha_1\left(c^{l,m,n}_{0,0,0}-c^{l-1,m,n}_{0,0,0}\right)
+ *     \right),\\
+ * \tilde{c}^{l,m,n}_{0,1,0}=\mathrm{minmod}
+ *   &\left(c_{0,1,0}^{l,m,n},
+ *          \alpha_1\left(c^{l,m+1,n}_{0,0,0}-c^{l,m,n}_{0,0,0}\right),
+ *          \alpha_1\left(c^{l,m,n}_{0,0,0}-c^{l,m-1,n}_{0,0,0}\right)
+ *     \right),\\
+ * \tilde{c}^{l,m,n}_{0,0,1}=\mathrm{minmod}
+ *   &\left(c_{0,0,1}^{l,m,n},
+ *          \alpha_1\left(c^{l,m,n+1}_{0,0,0}-c^{l,m,n}_{0,0,0}\right),
+ *          \alpha_1\left(c^{l,m,n}_{0,0,0}-c^{l,m,n-1}_{0,0,0}\right)
+ *     \right),
```

Comment:
(a) did you type all of this? 😄 and (b) would a shorter example suffice, perhaps in 2d?

Reply:
- yes
- no, I found the 2d example insufficient.